chef/data_bags/crowbar/migrate/network/011_add_ovs_attributes.rb

```diff
@@ -1,17 +1,33 @@
 def upgrade(ta, td, a, d)
-  a["networks"]["nova_floating"]["add_ovs_bridge"] = ta["networks"]["nova_floating"]["add_ovs_bridge"]
-  a["networks"]["nova_floating"]["bridge_name"] = ta["networks"]["nova_floating"]["bridge_name"]
-  a["networks"]["nova_fixed"]["add_ovs_bridge"] = ta["networks"]["nova_fixed"]["add_ovs_bridge"]
-  a["networks"]["nova_fixed"]["bridge_name"] = ta["networks"]["nova_fixed"]["bridge_name"]
+  unless a["networks"]["nova_floating"].has_key? "add_ovs_bridge"
+    a["networks"]["nova_floating"]["add_ovs_bridge"] = ta["networks"]["nova_floating"]["add_ovs_bridge"]
+  end
+  unless a["networks"]["nova_floating"].has_key? "bridge_name"
+    a["networks"]["nova_floating"]["bridge_name"] = ta["networks"]["nova_floating"]["bridge_name"]
+  end
+  unless a["networks"]["nova_fixed"].has_key? "add_ovs_bridge"
+    a["networks"]["nova_fixed"]["add_ovs_bridge"] = ta["networks"]["nova_fixed"]["add_ovs_bridge"]
+  end
+  unless a["networks"]["nova_fixed"].has_key? "bridge_name"
+    a["networks"]["nova_fixed"]["bridge_name"] = ta["networks"]["nova_fixed"]["bridge_name"]
+  end
 
   return a, d
 end
 
 def downgrade(ta, td, a, d)
-  a["networks"]["nova_floating"].delete "add_ovs_bridge"
-  a["networks"]["nova_floating"].delete "bridge_name"
-  a["networks"]["nova_fixed"].delete "add_ovs_bridge"
-  a["networks"]["nova_fixed"].delete "bridge_name"
+  unless ta["networks"]["nova_floating"].has_key? "add_ovs_bridge"
+    a["networks"]["nova_floating"].delete "add_ovs_bridge"
+  end
+  unless ta["networks"]["nova_floating"].has_key? "bridge_name"
+    a["networks"]["nova_floating"].delete "bridge_name"
+  end
+  unless ta["networks"]["nova_fixed"].has_key? "add_ovs_bridge"
```

Comment:
Style/DeprecatedHashMethods: `Hash#has_key?` is deprecated in favor of `Hash#key?`. (https://github.com/bbatsov/ruby-style-guide#hash-key)